### PR TITLE
[For CI] Random property/suffix unit tests, also more DBC std::array-fication

### PR DESCRIPTION
**Pull request by Treeston**

### CODE REVIEW

File: src/server/game/AuctionHouse/AuctionHouseMgr.cpp [hunk 1]
```diff
@@ -817,7 +817,7 @@ void AuctionHouseObject::BuildListAuctionItems(WorldPacket& data, Player* player
                     // Append the suffix (ie: of the Monkey) to the name using localization
                     // or default enUS if localization is invalid
                     name += ' ';
-                    name += suffix[locdbc_idx >= 0 ? locdbc_idx : LOCALE_enUS];
+                    name += suffix->operator[](locdbc_idx >= 0 ? locdbc_idx : LOCALE_enUS);
```
Comment (Shauren) [inline src/server/game/AuctionHouse/AuctionHouseMgr.cpp:820]: yeah no

(*suffix)

File: src/server/game/Chat/HyperlinkTags.cpp [hunk 2]
```diff
@@ -113,9 +113,47 @@ bool Trinity::Hyperlinks::LinkTags::item::StoreTo(ItemLinkData& val, std::string
...
+        else
+            return false;
+    }
+    else
+    {
+        val.RandomSuffix = nullptr;
+        val.RandomProperty = nullptr;
+    }
+
+    if (static_cast<bool>(val.RandomSuffix) != static_cast<bool>(val.RandomSuffixBaseAmount))
```
Comment (Shauren) [inline src/server/game/Chat/HyperlinkTags.cpp:153]: what

File: src/server/game/Chat/Hyperlinks.cpp [hunk 3]
```diff
@@ -157,8 +147,9 @@ struct LinkValidator<LinkTags::item>
                 continue;
             if (randomSuffixes)
             {
-                std::string_view randomSuffix(randomSuffixes[i]);
+                std::string_view randomSuffix(randomSuffixes->operator[](i));
```
Comment (Shauren) [inline src/server/game/Chat/Hyperlinks.cpp:150]: (*)

File: src/server/game/Chat/HyperlinkTags.cpp (hunk 2 above)
Comment (Treeston) [reply src/server/game/Chat/HyperlinkTags.cpp:153]: one needs to be present if and only if the other is

Comment (Shauren) [reply src/server/game/Chat/HyperlinkTags.cpp:153]: rewrite with explicit != nullptr/0, these casts are just horrible

**Pull request merged**